propolis/src/vmm/machine.rs

```diff
@@ -328,18 +364,47 @@ impl<'a> MemCtx<'a> {
             let req_offset = start - addr;
             match ent.kind {
                 MapKind::SysMem(_, prot) | MapKind::Rom(_, prot) => {
-                    // TODO: This protection check may be redundant with
-                    // mapping, which has its own memory protection tracking.
-                    if prot.contains(need_prot) {
-                        let base = ent.guest_map.as_ref().unwrap().as_ref();
-                        return base.subregion(req_offset, len);
-                    }
+                    let guest_map = ent
```

Comment:
It's a stylistic nit, but I felt like the `base` local was slightly more readable. There's a big chain of methods now that's a tad rough to keep track of.